src/actions.rs

```diff
@@ -0,0 +1,886 @@
+#[derive(Debug, Clone)]
+pub enum PointerOrigin {
+    /// Coordinates are relative to the top-left corner of the browser window.
+    Viewport,
+    /// Coordinates are relative to the pointer's current position.
+    Pointer,
+    /// Coordinates are relative to the specified element's center position.
+    WebElement(Element),
+}
+
+impl From<PointerOrigin> for WDActions::PointerOrigin {
+    fn from(po: PointerOrigin) -> Self {
+        match po {
+            PointerOrigin::Viewport => WDActions::PointerOrigin::Viewport,
+            PointerOrigin::Pointer => WDActions::PointerOrigin::Pointer,
+            PointerOrigin::WebElement(e) => WDActions::PointerOrigin::Element(
+                webdriver::common::WebElement(e.element_id().to_string()),
+            ),
+        }
+    }
+}
+
+/// The type of pointer.
+#[derive(Debug, Clone)]
+pub enum PointerActionType {
+    /// A mouse pointer device.
+    Mouse,
+    /// A pen device.
+    Pen,
+    /// A touch device.
+    Touch,
+}
+
+impl From<PointerActionType> for WDActions::PointerType {
+    fn from(pat: PointerActionType) -> Self {
+        match pat {
+            PointerActionType::Mouse => WDActions::PointerType::Mouse,
+            PointerActionType::Pen => WDActions::PointerType::Pen,
+            PointerActionType::Touch => WDActions::PointerType::Touch,
+        }
+    }
+}
+
+impl From<PointerAction> for WDActions::PointerActionItem {
+    fn from(pa: PointerAction) -> Self {
+        match pa {
+            PointerAction::Pause { duration } => WDActions::PointerActionItem::General(
+                WDActions::GeneralAction::Pause(WDActions::PauseAction {
+                    duration: Some(duration),
+                }),
+            ),
+            PointerAction::Down { button } => WDActions::PointerActionItem::Pointer(
+                WDActions::PointerAction::Down(WDActions::PointerDownAction { button }),
+            ),
+            PointerAction::Up { button } => WDActions::PointerActionItem::Pointer(
+                WDActions::PointerAction::Up(WDActions::PointerUpAction { button }),
+            ),
+            PointerAction::Move {
+                duration,
+                origin,
+                x,
+                y,
+            } => WDActions::PointerActionItem::Pointer(WDActions::PointerAction::Move(
+                WDActions::PointerMoveAction {
+                    duration: Some(duration),
+                    origin: origin.into(),
+                    x: Some(x),
+                    y: Some(y),
+                },
+            )),
+            PointerAction::Cancel => {
+                WDActions::PointerActionItem::Pointer(WDActions::PointerAction::Cancel)
+            }
+        }
+    }
+}
+
+/// A channel containing `Null` actions.
+#[derive(Debug, Clone)]
+pub struct NullActionChannel {
+    /// An identifier to distinguish this channel from others. Choose a meaningful string.
+    /// May be useful for debugging.
```

Comment:
```suggestion
    /// An identifier to distinguish this channel from others.
    ///
    /// Choose a meaningful string as it may be useful for debugging.
```
Is going to render more nicely.